packages/flutter_tools/test/commands.shard/hermetic/android_analyze_test.dart

```diff
@@ -130,3 +149,30 @@ class FakeAndroidBuilder extends Fake implements AndroidBuilder {
     return outputPath;
   }
 }
+
+class FakeAndroidAnalyzeCommand extends AnalyzeCommand {
```

Comment:
Rather than creating this abstraction, could you just run the command first passing the args you want, and then call `command.shouldRunPub`?

Reply:
There isn't a way to pass args without running the command since the logic is hardcoded in args package. Running the command will throw because the throwing_pub.dart

Reply:
You can provide a FakePub to `testUsingContext`, but this is fine.